retry/retry.go

```diff
@@ -2,9 +2,27 @@ package retry
 
 import (
 	"context"
+	"fmt"
+	"math"
 	"time"
 )
 
+type RetryErrScope string
+
+const (
+	MaxDurationScope RetryErrScope = "max-duration"
+	ContextScope     RetryErrScope = "context"
+)
+
+type RetryErr struct {
```

Comment:
Following this golang doc (https://github.com/golang/go/wiki/Errors#naming), you should name this:
```suggestion
type RetryError struct {
```

Reply:
Done